src/ALCProxy/ProxyBuilder.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.Loader;
+using HelloWorld;
+
+namespace ALCProxy
+{
+
+    public class ProxyBuilder <T, I> where T : I
+    {
+
+        private readonly AssemblyLoadContext _alc;
+        private readonly string _assemblyPath;
+        private readonly Assembly _a;
+        public object instanceObj;
+
+
+        public ProxyBuilder(AssemblyLoadContext alc, string assemblyPath)
+        {
+            _alc = alc;
+            _assemblyPath = assemblyPath;
+            _a = _alc.LoadFromAssemblyPath(_assemblyPath);
+        }
+
+
+        private void CreateInstanceObj()
+        {
+            Type t = _a.GetType(_a.GetName().Name + "." + typeof(T).Name);
```

Comment:
I don't believe it's guaranteed that the assembly name matches the namespace of the type (there's additionally a way to specify a type in a particular assembly. For example `System.IO.File,System.IO.FileSystem` refers to the type `File` in the `System.IO` namespace from the `System.IO.FileSystem` assembly). The way this works with CreateInstanceAndUnwrap is that the type is specified as a string including the namespace of the type but not the assembly. Eventually it would be good to include some tests to make sure this works correctly.